src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -14,95 +14,95 @@ import { updatePlotConfig } from '../../../redux/actions/componentConfig';
 import ReorderableList from '../../ReorderableList';
 import { getCellSetsHierarchy } from '../../../redux/selectors';
 
+const convertToReorderableListData = (cellClassKeys, selected, hierarchy) => (
+  cellClassKeys.map((key) => {
+    const { name } = _.find(hierarchy, ({ key: currentKey }) => currentKey === key);
+
+    return {
+      key, disabledReorder: !selected, name,
+    };
+  })
+);
+
 const HeatmapMetadataTrackSettings = (props) => {
   const dispatch = useDispatch();
 
   const { componentType } = props;
-  const selectedHierarchy = useSelector(getCellSetsHierarchy());
+  const hierarchy = useSelector(getCellSetsHierarchy());
 
   const selectedTracks = useSelector(
     (state) => state.componentConfig[componentType].config.selectedTracks,
   );
 
-  const getTrackData = () => selectedHierarchy.map(
-    (data) => ({ selected: selectedTracks.includes(data.key), key: data.key }),
-  );
+  const [listData, setListData] = useState([]);
 
-  const isInitialRenderRef = useRef(true);
+  useEffect(() => {
```

Comment:
I noticed that this `useEffect` gets triggered twice every time `componentConfig/update` action happens (changing `selectedTracks` is effectively a change of `componentConfig`), which doesn't seem right. Shouldn't the useEffect get triggered only once?

Reply:
@cosa65 did you manage to figure this out?

Reply:
I've been looking into it and it seems that `HeatmapMetadataTrackSettings` is being rendered twice (as in there are two components that are `HeatmapMetadataTrackSettings`), I am not sure exactly how yet so I will take a look at the parent components of this

Reply:
It seems this is the reason for the issue: 
![image](https://user-images.githubusercontent.com/8465903/148101423-c9125a8f-a4de-4091-86bc-83ac2eab64c0.png)

after wrapping the items we show within <menu.item> components it doesn't render twice, however the menu.item changes some of the menu's behavior so I will have to look into that now

Reply:
In the end for the solution I implemented a `Menu.Item` wrapper that removes all custom styling and behavior from it so that the metadata tracks and group by settings components don't change anything (except that this time there's only one of each instead of duplicates).

Reply:
Great find! Can you make sure that you show this to the rest of the team, particularly Agi and Stefan, to avoid them struggling with this in the future?